lib/iris/_deprecation_helpers.py

```diff
@@ -24,10 +24,21 @@
 
 import warnings
 
+from iris.exceptions import IrisError
+
+
+class IrisDeprecation(UserWarning):
+    """An Iris deprecation warning."""
+    pass
+
+
+def warn_deprecated(msg, **kwargs):
```

Comment:
You will be glad to know we are not treading new territory here (http://matplotlib.org/api/cbook_api.html#matplotlib.cbook.MatplotlibDeprecationWarning). However, it may be sensible to provide more functionality in our "deprecation" function.

Matplotlib makes this somewhat repeatable with its [```deprecated(since, message='', name='', alternative='', pending=False, obj_type='function')``` function](http://matplotlib.org/api/cbook_api.html#matplotlib.cbook.deprecated)